### PR TITLE
Allow field labels to work for routes

### PR DESCRIPTION
**Pull request by smarterclayton**
Fixes bug 1261452

### CODE REVIEW

**Comment by smarterclayton** [comment]
[test]

**Comment by smarterclayton** [comment]
@derekwaynecarr

**Comment by smarterclayton** [comment]
kubernetes/kubernetes#13746

**Comment by smarterclayton** [comment]
[merge] because no one ever looks at my code when I ask them to...

File: Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go [hunk 1]
```diff
@@ -476,11 +476,9 @@ func (e *Etcd) WatchPredicate(ctx api.Context, m generic.Matcher, resourceVersio
 	}
 
 	if name, ok := m.MatchesSingle(); ok {
-		key, err := e.KeyFunc(ctx, name)
-		if err != nil {
-			return nil, err
+		if key, err := e.KeyFunc(ctx, name); err == nil {
+			return e.Storage.Watch(key, version, filterFunc)
```
Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go]: Why ignore the error?

Comment (smarterclayton) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go]: The scenario "do we match a single entry" is only valid when there is a valid key.  So if you get an error, it means you can't watch on the provided thing.

Comment (liggitt) [inline Godeps/_workspace/src/k8s.io/kubernetes/pkg/registry/generic/etcd/etcd.go]: ok, watching the single thing is an optimization, so we fall back to watching the list

**Pull request merged**